api/provider/aws/retry_test.go

```diff
@@ -0,0 +1,30 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRetry_Timeout(t *testing.T) {
+	fn := func() error {
+		return fmt.Errorf("Tick")
+	}
+
+	if err := retry(20*time.Millisecond, 5*time.Millisecond, fn); err != nil {
+		assert.Equal(t, err, errors.New(errors.FailedRequestTimeout, nil))
```

Comment:
why do we assume we should get a timeout in this test? There's nothing blocking in `fn` that would cause a timeout to occur, so it seems like this is either passing because:
a) it's relying on the time it takes for goroutines to spin-up and `fn` to execute to be less than 20ms, which is a very brittle assumption. 
b) or, the logic in `retry` is broken and returns timeout errors when it actually shouldn't.

Reply:
Hmm, my intended goal was to have it retry when there is an error. Since FN only returns an error the function times out before getting a success or nil return.